### PR TITLE
softplus_superset

### PR DESCRIPTION
**Pull request by Ishticode**
Based on the additional arguments in [torch softplus](https://pytorch.org/docs/stable/generated/torch.nn.functional.softplus.html).

This just requires that the `x` is `x * beta` and that the final results is multiplied by `1/beta` as clear from the formula.

`torch` takes this as arg while for other backends it is done manually. Test passing.

Should we consider adding the argument `threshold` also from [torch softplus](https://pytorch.org/docs/stable/generated/torch.nn.functional.softplus.html)**?**

### CODE REVIEW

**Review by djl11 — CHANGES_REQUESTED** [review]
looks great! Yes please add the threshold also and then request another review :)

**Comment by Ishticode** [comment]
> looks great! Yes please add the threshold also and then request another review :)

added with docstring examples.

**Review by djl11 — CHANGES_REQUESTED** [review]
it says values above the threshold "revert to a linear function", you seem to just be clipping the values at `20`, I believe this is incorrect.

**Comment by Ishticode** [comment]
> it says values above the threshold "revert to a linear function", you seem to just be clipping the values at `20`, I believe this is incorrect.

Sorry my bad. Updated now. Somehow it went thru tests with 50 example set. The current implementatio is according to [**this**](https://github.com/pytorch/pytorch/blob/2386cd2945498ce7261b761a8d9bd5b59d06c5a1/torch/_refs/nn/functional/__init__.py#L243). It seems need to add support for None as well.

**Review by djl11 — CHANGES_REQUESTED** [review]
the `threshold` argument should also have a default of `None` in each case. Please see [this section](https://github.com/unifyai/ivy/blob/b467afb70ead6ea2a826910a22654c16f9a31e8a/docs/partial_source/deep_dive/2_superset_behaviour.rst#balancing-generalization-with-efficiency) of the Deep Dive for a discussion on this :)

**Comment by Ishticode** [comment]
Could you take a special look at torch backend. It doesn't accept None for either beta or threshold so in case these args are passed the default setting are used.

**Review by djl11 — CHANGES_REQUESTED** [review]
I reduced the number of lines in the torch implementation, but otherwise is good to merge!

File: ivy/functional/backends/torch/activations.py [hunk 1]
```diff
@@ -70,8 +70,20 @@ def softmax(
...
 
-def softplus(x: torch.Tensor, /, *, out: Optional[torch.Tensor] = None) -> torch.Tensor:
-    return torch.nn.functional.softplus(x)
+def softplus(x: torch.Tensor,
+             /,
+             *,
+             beta: Optional[Union[int, float]] = None,
+             threshold: Optional[Union[int, float]] = None,
+             out: Optional[torch.Tensor] = None) -> torch.Tensor:
+    if beta is None and threshold is None:
```
[the remark this answers is not in the document]
Comment (Ishticode) [reply ivy/functional/backends/torch/activations.py:79]: very cool @djl11 🙂.

**Pull request merged**